packages/web/src/components/SignUpForm/SignUpForm.tsx

```diff
@@ -16,29 +16,101 @@ type Props = {
 
 export const SignUpForm = ({ isLoading, error, onSubmit }: Props) => {
   const [firstName, setFirstName] = useState('');
+  const [firstNameError, setFirstNameError] = useState('');
+  const [firstNameTouched, setFirstNameTouched] = useState(false);
+
   const [lastName, setLastName] = useState('');
+  const [lastNameError, setLastNameError] = useState('');
+  const [lastNameTouched, setLastNameTouched] = useState(false);
+
   const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [emailTouched, setEmailTouched] = useState(false);
+
   const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [passwordTouched, setPasswordTouched] = useState(false);
+
   const [passwordMatch, setPasswordMatch] = useState('');
+  const [passwordMatchError, setPasswordMatchError] = useState('');
+  const [passwordMatchTouched, setPasswordMatchTouched] = useState(false);
+
   const [isValid, setIsValid] = useState(false);
-  const [isPasswordMatch, setIsPasswordMatch] = useState(true);
+
+  // I've decided to keep it here for now, but later probably we can use 'validate' or/and 'Formik' npm libs.
+  const validateEmail = (emailToTest) => {
+    // eslint-disable-next-line
+    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(String(emailToTest).toLowerCase());
+  };
 
   useEffect(() => {
     setIsValid(
-      !!email && !!password && !!passwordMatch && !!firstName && !!lastName
+      !emailError &&
+        !passwordError &&
+        !passwordMatchError &&
+        !firstNameError &&
+        !lastNameError
     );
-  }, [email, password, passwordMatch, firstName, lastName]);
+  }, [
+    emailError,
+    passwordError,
+    passwordMatchError,
+    firstNameError,
+    lastNameError,
+  ]);
 
+  // validation and error setting for First Name
   useEffect(() => {
-    setIsPasswordMatch(true);
-  }, [password, passwordMatch]);
+    if (firstName.length <= 3) {
+      setFirstNameError('First name has to be greater than 3 symbols.');
+    } else {
+      setFirstNameError('');
+    }
+  }, [firstName]);
 
-  const handleSubmit = (e: FormEvent) => {
-    e.preventDefault();
+  // validation and error setting for Last Name
+  useEffect(() => {
+    if (lastName.length <= 3) {
+      setLastNameError('Last name has to be greater than 3 symbols.');
+    } else {
+      setLastNameError('');
+    }
+  }, [lastName]);
+
+  // validation and error setting for Email
+  useEffect(() => {
+    if (!validateEmail(email)) {
+      setEmailError('Email is invalid.');
+    } else {
+      setEmailError('');
+    }
+  }, [email]);
+
+  // validation and error setting for Password
+  useEffect(() => {
+    if (password.length < 8) {
+      setPasswordError('Password has to be greater than 8 symbols.');
```

Comment:
I'd suggest move all errors to a separate file. It will help to easier localize app without code changes

Reply:
Probably later we will start using formik or other tools, to handle form validation. So in my opinion due to lack of time I would suggest skip this for now.